### ISSUE TITLE
Program Page: support for tables

### ISSUE THREAD

**Issue by pdpinch** [mitodl/micromasters#4194]
As a member of a course team, I'd like to be able to add and edit tables on my program tab page (e.g. https://micromasters.mit.edu/dedp/pathways-masters/), and have it display as HTML. This will help accessibility, SEO and mobile rendering, and also make the content easier to maintain. 

The default table styling should be similar to [this graphic](https://do5zh7b0lqdye.cloudfront.net/images/schedule.width-800.png) embedded in the [POM program schedule page](https://micromasters.mit.edu/pom/program-schedule/) (_not_ the timeline/calendar).

Wagtail supports a [table block module](http://docs.wagtail.io/en/v2.3/reference/contrib/table_block.html). We may need to change the field type of the content field in the program tab page type to a streamfield to use this. 

At this time, we are expecting there to be text and links in the table.

**Comment by pdpinch** [comment]
I updated the description to add a link to an example page that we want to change and to make it clearer that the priority is adding a table to the "Program Tab" page type, in the content field. It would be nice to also allow tables in the Description field of the "Program Page" page type, but that isn't the current priority.

**Comment by tasawernawaz** [comment]
@pdpinch When I click on add child page, I see following page 
![screen shot 2019-01-11 at 11 40 14 am](https://user-images.githubusercontent.com/4245618/51017382-bb685c80-1595-11e9-9e6a-ff07e382ae26.png)

I believe this is a "Program Page" page type, I can't see anything like "Program Tab" page type.

**Comment by pdpinch** [comment]
I think you need to create a program page before you can add a child page to it. I'll send you a video.

**Comment by tasawernawaz** [comment]
thanks, that will surely help.

On Jan 11, 2019 8:44 PM, "Peter Pinch" <notifications@github.com> wrote:

I think you need to create a program page before you can add a child page
to it. I'll send you a video.

—
You are receiving this because you were assigned.

Reply to this email directly, view it on GitHub
<https://github.com/mitodl/micromasters/issues/4194#issuecomment-453559490>,
or mute the thread

**Comment by pdpinch** [comment]
Here's a video walking through the steps for creating a program page and a program tab page: https://share.vidyard.com/watch/dLLHdsc7R6syutDLE2t1ct?

**Comment by tasawernawaz** [comment]
@pdpinch thanks for the video walk through, I was able to create program tab page.

I have gone through the documentation of [table block](http://docs.wagtail.io/en/v2.3/reference/contrib/table_block.html) which we can use by using [Streamfield](http://docs.wagtail.io/en/v2.0/topics/streamfield.html). In stream fields documentation I also found a block type called Raw [HTML block](http://docs.wagtail.io/en/v2.0/topics/streamfield.html#rawhtmlblock), which could be useful here to render HTML (including tables and other html stuff too).

One more thing, we can not simply migrate from rich text field to stream field, we have to create data migrations for backward compatibility from rich text fields to stream fields [streamfield-migrating-richtext](http://docs.wagtail.io/en/v2.0/topics/streamfield.html#streamfield-migrating-richtext).

Before moving forward I want to get your opinion about using HTML Block vs Table Block.
